addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -1976,6 +1976,9 @@ export class PosStore extends Reactive {
 
         // Add the product after having the extra information.
         await this.addProductFromUi(product, options);
+        if (product.tracking == "serial") {
+            await this.selectedOrder.selected_orderline.set_quantity_by_lot();
```

Comment:
```suggestion
            this.selectedOrder?.selected_orderline?.set_quantity_by_lot();
```

May I know why you're awaiting it?